test/e2e/framework/daemonsetjobs_utils.go

```diff
@@ -130,24 +129,34 @@ func HOIsDaemonSetJobStarted(workflowClient versioned.Interface, daemonsetjob *d
 	}
 }
 
+func displayJob(job *batch.Job) error {
+	fmt.Printf("Job: %#v\n", job.GetObjectMeta())
+	return nil
+}
+
 // HOIsDaemonSetJobJobsStarted is an higher order func that returns the func that checks whether Jobs linked to a DaemonSetJob are started
 func HOIsDaemonSetJobJobsStarted(kubeclient clientset.Interface, daemonsetjob *daemonsetjobv1.DaemonSetJob) func() error {
 	return func() error {
+		labelSelector := controller.InferDaemonSetJobLabelSelectorForJobs(daemonsetjob).String()
 		jobs, err := kubeclient.BatchV1().Jobs(daemonsetjob.Namespace).List(metav1.ListOptions{
-			LabelSelector: labels.Set(daemonsetjob.Labels).AsSelector().String(),
+			LabelSelector: labelSelector,
```

Comment:
@clamoriniere1A may you have a look at this and tell me if this looks good to you?
Thanks

Reply:
Yes thanks, for fixing this issue!